src/python/pants/engine/internals/graph.py

```diff
@@ -635,7 +635,8 @@ async def find_all_targets() -> AllTargets:
     tgts = await Get(
         Targets,
         RawSpecsWithoutFileOwners(
-            recursive_globs=(RecursiveGlobSpec(""),), description_of_origin="the `AllTargets` rule"
+            recursive_globs=(RecursiveGlobSpec(""),), description_of_origin="the `AllTargets` rule",
+            unmatched_glob_behavior=GlobMatchErrorBehavior.warn,
```

Comment:
This was necessary to prevent failure of the existing tests in `export_test.py` due to no targets being defined.

Reply:
```
native_engine.IntrinsicError: Unmatched glob from the `AllTargets` rule: "**"
```

https://github.com/pantsbuild/pants/actions/runs/9311082311/job/25629746990#step:11:604

Reply:
Looks like this is back to being an `error`, so why not fall back to the default like before?

Reply:
I had set it to warn originally since it (seemed to) fixed a test failure in `export_test.py`, but if CI passes as is then I agree I should just remove the change. Then I will forever wonder what the state of my tree had been earlier in development to cause tests in `export_test.py` to fail on me.

Reply:
I removed it. Will see what CI decides.